src/pokemon_summary_screen.c

```diff
@@ -0,0 +1,3680 @@
+        gMain.state++;
+        break;
+    default:
+        SetVBlankCallback(sub_81BFAD0);
+        SetMainCallback2(sub_81BFAB4);
+        return TRUE;
+    }
+    return FALSE;
+}
+
+void sub_81BFE24()
+{
+    ResetBgsAndClearDma3BusyFlags(0);
+    InitBgsFromTemplates(0, &gUnknown_0861CBB4, 4);
+    SetBgTilemapBuffer(1, &gUnknown_0203CF1C->unkTilemap2);
+    SetBgTilemapBuffer(2, &gUnknown_0203CF1C->unkTilemap1);
+    SetBgTilemapBuffer(3, &gUnknown_0203CF1C->unkTilemap0);
+    sub_8121E10();
+    schedule_bg_copy_tilemap_to_vram(1);
+    schedule_bg_copy_tilemap_to_vram(2);
+    schedule_bg_copy_tilemap_to_vram(3);
+    SetGpuReg(0, 0x1040);
+    SetGpuReg(0x50, 0);
+    ShowBg(0);
+    ShowBg(1);
+    ShowBg(2);
+    ShowBg(3);
+}
+
+u8 sub_81BFEB0()
+{
+    switch (gUnknown_0203CF1C->unk40F0)
+    {
+    case 0:
+        reset_temp_tile_data_buffers();
+        decompress_and_copy_tile_data_to_vram(1, &gUnknown_08D97D0C, 0, 0, 0);
+        gUnknown_0203CF1C->unk40F0++;
+        break;
+    case 1:
+        if (free_temp_tile_data_buffers_if_possible() != 1)
+        {
+            LZDecompressWram(&gUnknown_08D9862C, gUnknown_0203CF1C->unkTilemap0);
+            gUnknown_0203CF1C->unk40F0++;
+        }
+        break;
+    case 2:
+        LZDecompressWram(&gUnknown_08D98CC8, gUnknown_0203CF1C->unkTilemap0_1);
+        gUnknown_0203CF1C->unk40F0++;
+        break;
+    case 3:
+        LZDecompressWram(&gUnknown_08D987FC, gUnknown_0203CF1C->unkTilemap1_1);
+        gUnknown_0203CF1C->unk40F0++;
+        break;
+    case 4:
+        LZDecompressWram(&gUnknown_08D9898C, gUnknown_0203CF1C->unkTilemap2_1);
+        gUnknown_0203CF1C->unk40F0++;
+        break;
+    case 5:
+        LZDecompressWram(&gUnknown_08D98B28, gUnknown_0203CF1C->unkTilemap3_1);
+        gUnknown_0203CF1C->unk40F0++;
+        break;
+    case 6:
+        LoadCompressedPalette(&gUnknown_08D9853C, 0, 0x100);
+        LoadPalette(&gUnknown_08D85620, 0x81, 0x1E);
+        gUnknown_0203CF1C->unk40F0++;
+        break;
+    case 7:
+        LoadCompressedObjectPic(&gUnknown_0861CFBC);
+        gUnknown_0203CF1C->unk40F0++;
+        break;
+    case 8:
+        LoadCompressedObjectPic(&gUnknown_0861D074);
+        gUnknown_0203CF1C->unk40F0++;
+        break;
+    case 9:
+        LoadCompressedObjectPic(&gUnknown_0861D0F8);
+        gUnknown_0203CF1C->unk40F0++;
+        break;
+    case 10:
+        LoadCompressedObjectPalette(&gUnknown_0861D100);
+        gUnknown_0203CF1C->unk40F0++;
+        break;
+    case 11:
+        LoadCompressedObjectPalette(&gUnknown_0861D07C);
+        gUnknown_0203CF1C->unk40F0++;
+        break;
+    case 12:
+        LoadCompressedPalette(&gUnknown_08D97B84, 0x1D0, 0x60);
+        gUnknown_0203CF1C->unk40F0 = 0;
+        return 1;
+    }
+    return 0;
+}
+
+void sub_81C0098(struct Pokemon *poke)
+{
+    if (gUnknown_0203CF1C->unk40BD == 0)
+    {
+        struct Pokemon *pokeMons = gUnknown_0203CF1C->unk0->mons;
```

Comment:
This should be named something more descriptive, like partyMon if they're in the party.